sorana/src/main/java/org/java/mentorship/service/TestService.java

```diff
@@ -0,0 +1,74 @@
+package org.java.mentorship.service;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import org.java.mentorship.domain.Song;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+@Service
+@AllArgsConstructor
+public class TestService {
+
+    private List<Song> songs;
+
+    public List<Song> getSongs(){
+        return this.songs;
+    }
+
+    public List<Song> getSongsByStyle(String style){
+
+        if(style == null) return this.songs;
+
+        List<Song> songsByStyle = new ArrayList<>();
+
+        songsByStyle.stream()
+                .filter(song -> song.getStyle().equals(style))
+                .collect(Collectors.toList());
+        return songsByStyle;
+    }
+
+    public Song getSongById(int id){
+        if(id == 0) return null;
```

Comment:
Nice defensive coding.

Indeed, we should have some validations on the inputs from the user.
Never trust the user 😄